lib/spack/spack/util/compression.py

```diff
@@ -615,6 +710,7 @@ def strip_last_extension(path):
 def strip_extension(path, ext=None):
     """Get the part of a path that does not include its compressed
     type extension."""
+    path = expand_contracted_extension_in_path(path)
```

Comment:
I think I meant to put this logic in `strip_single_extension`

Reply:
Yes, I pretty clearly meant to put this in `strip_last_extension` and convert the decompression methods to use that instead of `strip_last_extension` like I did for the 7zip method. Amended this way locally.